test/unit/product/product-component.js

```diff
@@ -401,66 +410,183 @@ describe('Product Component class', () => {
+        describe('if button destination is cart', () => {
+          let addToCartStub;
+
+          beforeEach(() => {
+            product.config.product.buttonDestination = 'cart';
+            addToCartStub = sinon.stub(product.cart, 'addVariantToCart');
+          });
+
+          it('closes modal', () => {
+            product.onButtonClick(evt, target);
+            assert.calledOnce(closeModalSpy);
+          });
+
+          it('calls userEvent with addVariantToCart', () => {
+            product.onButtonClick(evt, target);
+            assert.calledOnce(userEventStub);
+            assert.calledWith(userEventStub, 'addVariantToCart');
+          });
+
+          it('tracks addVariantToCart', () => {
+            product.onButtonClick(evt, target);
+            assert.calledOnce(trackMethodStub);
+            assert.calledWith(trackMethodStub, sinon.match.func, 'Update Cart', product.selectedVariantTrackingInfo);
+            assert.calledOnce(addToCartStub);
+            trackMethodStub.getCall(0).args[0]();
+            assert.calledTwice(addToCartStub);
+          });
+
+          it('adds variant to cart with the right quantity of selected variant', () => {
+            product.selectedQuantity = 1111;
+
+            product.onButtonClick(evt, target);
+            assert.calledOnce(addToCartStub);
+            assert.calledWith(addToCartStub, product.selectedVariant, 1111);
+            addToCartStub.restore();
+          });
+
+          it('sets target to active el if iframe exists', () => {
+            product.iframe = {};
+            product.onButtonClick(evt, target);
+            assert.calledOnce(setActiveElSpy);
+            assert.calledWith(setActiveElSpy, target);
           });
+        });
+
+        describe('if button destination is modal', () => {
+          let openModalStub;
 
+          beforeEach(() => {
+            product.config.product.buttonDestination = 'modal';
+            openModalStub = sinon.stub(product, 'openModal');
+            product.onButtonClick(evt, target);
+          });
+
+          afterEach(() => {
+            openModalStub.restore();
+          });
+
+          it('sets active element to target', () => {
+            assert.calledOnce(setActiveElSpy);
+            assert.calledWith(setActiveElSpy, target);
+          });
+
+          it('opens modal', () => {
+            assert.calledOnce(openModalStub);
+          });
+        });
+
+        it('opens online store if button destination is online store', () => {
+          const openOnlineStoreStub = sinon.stub(product, 'openOnlineStore');
+          product.config.product.buttonDestination = 'onlineStore';
+          product.onButtonClick(evt, target);
+          assert.calledOnce(openOnlineStoreStub);
+          openOnlineStoreStub.restore();
+        });
+
+        describe('if button destination is checkout', () => {
+          let createCheckout;
+          let createCheckoutPromise;
           let addLineItems;
```

Comment:
Let's be consistent and keep the variable naming consistent when it comes to spies and stubs.